### PR TITLE
Automation for CEPH-83597243 - Test to verify RBD image with zero size

### PR DESCRIPTION
**Pull request by aarsharm**
# Description

<html>
<body>

Automation for  CEPH-83597243 - Test to verify RBD image with zero size
--

<br class="Apple-interchange-newline">
</body>
</html>
[CEPH-83597243 - Test to verify RBD image with zero size](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83597243)

Test logs :  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-5D34E7/Test_to_verify_RBD_image_with_zero_size_0.log

### CODE REVIEW

File: tests/rbd/test_rbd_image_zero_size.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+3. Create another image inside this pool with size 1024 (rbd create images/image_1024size --size 1024)
+4. Use rbd du command to check that image size for both images
+5. Resize second image size to 0
+6. Remove that images
+
+"""
+
+from ceph.parallel import parallel
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import (
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_image_zero_size.py:23]: `rbd_utils` got deprecated, can you please use `initial_rbd_config` from from `ceph.rbd.initial_config import initial_rbd_config`

File: tests/rbd/test_rbd_image_zero_size.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    
+    if kw.get("client_node"):
+        client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+    else:
+        client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+    rbd = Rbd(client)
+
+    if rbd_obj:
+        log.info("Executing test on Replication pool \n\n")
+        if resize_image_with_zero_size(
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_image_zero_size.py:106]: no need to pass generic `rbd` again here, can you please refer this test 
https://github.com/red-hat-storage/cephci/blob/master/tests/rbd/test_rbd_encryption.py#L16

File: tests/rbd/test_rbd_image_zero_size.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        return 0
+
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
+
+    finally:
+        log.info("Cleanup")
+        if pool_type == "ec_pool_config":
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_image_zero_size.py:81]: pool clean up needs to be done for both rep-pool and EC-pool, please use this cleanup from here https://github.com/red-hat-storage/cephci/blob/master/ceph/rbd/workflows/cleanup.py
also make sure irrespective of test pass or fail pool should be deleted.

File: tests/rbd/test_rbd_image_zero_size.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            log.info(f"Resize Image {image} to zero \n\n")
+            resize_config = {"image-spec": f"{pool}/{image}", "size": resize_to, "allow-shrink": True}
+            out, err = rbd.resize(**resize_config)
+            if "100% complete" not in out + err:
+                log.error(
+                    f"Image {image} resize to 0 failed with error {out} {err}"
+                )
+                return 1
+
+        log.info(f"Verify Image {image} size")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_image_zero_size.py:60]: we need to verify with `rbd du <pool_name>/<image_name>` make sure this command should not fail for 0 size images that is the intention of this test case, please make sure your validation for it.

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

**Comment by aarsharm** [comment]
Passing Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-9QTW0X/Test_to_verify_RBD_image_with_zero_size_0.log

File: suites/squid/rbd/tier-2_rbd_regression.yaml [hunk 5]
```diff
@@ -189,6 +189,31 @@ tests:
       name: Test to verify image resize operation while changing image feature
       polarion-id: CEPH-9861
 
+  - test:
+      desc: Verify RBD image with zero size
+      config:
+        resize_to: 0
+        rep_pool_config:
+          num_pools: 1
+          num_images: 2
```
Comment (sunilangadi2) [inline suites/squid/rbd/tier-2_rbd_regression.yaml:198]: since it's basic test please keep `num_images` also 1

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]
all please add these tests in `quincy and reef` test suites.

File: tests/rbd/test_rbd_image_zero_size.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+def run(**kw):
+    """Test to verify RBD image with zero size.
+    This test verifies images with zero size
+    Args:
+        kw: test data
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    """
+    log.info("CEPH-83597243 -  Test to verify RBD image with zero size \n\n")
```
[the remark this answers is not in the document]
Comment (aarsharm) [reply tests/rbd/test_rbd_image_zero_size.py]: Removed \n\n character

File: suites/squid/rbd/tier-2_rbd_regression.yaml (hunk 5 above)
Comment (aarsharm) [reply suites/squid/rbd/tier-2_rbd_regression.yaml:198]: As per steps mentioned in polarian: num_images is kept as 2 as we need to have 2 images, one created with size zero and another created with size 1024 later resized to zero.

File: suites/squid/rbd/tier-2_rbd_regression.yaml [hunk 7]
```diff
@@ -23,7 +23,7 @@
 
 tests:
 
-   #Setup the cluster
+  #Setup the cluster
```
[the remark this answers is not in the document]
Comment (aarsharm) [reply suites/squid/rbd/tier-2_rbd_regression.yaml]: Done

File: tests/rbd/test_rbd_image_zero_size.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        if resize_image_with_zero_size(
+            rbd, rbd_obj.get("rbd_reppool"), "rep_pool_config", **kw
+        ):
+            return 1
+        log.info("Executing test on EC pool \n\n")
+        if resize_image_with_zero_size(
+            rbd, rbd_obj.get("rbd_ecpool"), "ec_pool_config", **kw
+        ):
+            return 1
+    return 0
```
[the remark this answers is not in the document]
Comment (aarsharm) [reply tests/rbd/test_rbd_image_zero_size.py]: Done

File: tests/rbd/test_rbd_image_zero_size.py (hunk 1 above)
Comment (aarsharm) [reply tests/rbd/test_rbd_image_zero_size.py]: Replaced from tests.rbd.rbd_utils with "from ceph.rbd.initial_config import initial_rbd_config"

File: tests/rbd/test_rbd_image_zero_size.py (hunk 2 above)
Comment (aarsharm) [reply tests/rbd/test_rbd_image_zero_size.py]: Done

**Comment by aarsharm** [comment]
Reef pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4JX1YI
Quincy pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-11NVI5
Squid pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-5VVCN1

**Review by sunilangadi2 — APPROVED** [review]
Changes LGTM.

**Pull request merged**